### DISCUSSION TITLE
getServerSideProps is requesting an unknown api request is this a bug?

### DISCUSSION THREAD

**Discussion by Polgasol** [vercel/next.js#39708] [Help] [answered]
I'm using Express, Next, express-sessions. Everytime I make an api request, new sessions and cookies are being created . I debugged my express-sessions using ""start": DEBUG=express-session ts-node index.ts " in express this came out:
![Screenshot from 2022-08-18 10-19-29](https://user-images.githubusercontent.com/76750063/185284965-66dae8f0-51c9-4565-97f9-13f9dc78db09.png)
I'm only creating a simple api request from getServerSideProps
`const getData = async () => {
  const res = await axios.get('http://localhost:5000/api', { withCredentials: true });
  return res.data
}`

`export const getServerSideProps = async () => {
    const res = await getData();
    return {
      props: {
        res,
      },
    };
  } 
};`
Is this a bug from next?

**Comment by subhamX** [comment]
I think the cookies are in `req.cookies` and not `req.headers.cookie`. 

Reference: https://expressjs.com/en/5x/api.html#req.cookies

**Comment by gaston-flores** [comment, marked as answer]
I think you need to pass the cookies in `getServerSideProps`'s context to your API call. Add the session cookies as a parameter to your `getData` function and it should work.

```js
const getData = async req => {
  return axios.get('http://localhost:5000/api', {
    headers: {
      Cookie: req.headers.cookie
    },
    withCredentials: true
  });
}

export const getServerSideProps = async (context) => {
  const res = await getData(context.req);
  return {
    props: {
      res
    }
  };
};
```

**Reply by Polgasol** [reply]
Your answer solved the problem, the context part and the headers: {
      Cookie: req.headers.cookie
    }, solved the problem.
